eZ/Publish/Core/Persistence/Legacy/Notification/Gateway/DoctrineDatabase.php

```diff
@@ -0,0 +1,286 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Publish\Core\Persistence\Legacy\Notification\Gateway;
+
+use eZ\Publish\Core\Base\Exceptions\InvalidArgumentException;
+use eZ\Publish\Core\Persistence\Database\DatabaseHandler;
+use eZ\Publish\Core\Persistence\Database\InsertQuery;
+use eZ\Publish\Core\Persistence\Database\Query;
+use eZ\Publish\Core\Persistence\Database\SelectQuery;
+use eZ\Publish\Core\Persistence\Database\UpdateQuery;
+use eZ\Publish\SPI\Persistence\Notification\Handler;
+use eZ\Publish\SPI\Persistence\Notification\Notification;
+use PDO;
+
+class DoctrineDatabase implements Handler
+{
+    /** @var DatabaseHandler $handler */
+    protected $handler;
```

Comment:
You should use `\Doctrine\DBAL\Connection` instead of deprecated `DatabaseHandler`.